src/App/App.js

```diff
@@ -23,35 +17,27 @@ export default class App extends Component {
     this.state = {
       loggedIn: false,
       template: '',
-      // This attribute is temporary until API is functional
-      // stubCard: {
-      //       'projectTitle': 'Title1',
-      //       'repoName':     'RepoName1',
-      //       'createdAt':    'Created1',
-      //       'updatedAt':    'Updated1',
-      //   },
-      //   stubGithubCard: {
-      //     'cardTitle':       'This is the title',
-      //     'cardDescription': 'This is lots of long text from the body of my issue',
-      //     'cardNumber':       13,
-      //     'cardURL':         'https://github.com/JoelSmith123/git_wired_fe/issues/23',
-      //     'cardStatus':      'Open',
-      //     'cardColumn':      'Custom Column',
-      //   },
       viewProfile: false
     }
   }
 
+  componentDidMount() {
+    if (Boolean(this.props.user.getURLToken())) {
+      this.props.user.login()
+      this.setState({ loggedIn: this.props.user.isLoggedIn() })
+    }
+  }
+
   selectTemplate = (event, template) => {
     event.preventDefault()
 
     this.setState({ template })
   }
 
   changeLoggedInState = (event) => {
-    event.preventDefault()
+    if (event) {event.preventDefault()}
```

Comment:
@JoelSmith123 
I'm not very sure what this does, and so when I invoked this in a twice-nested component, it wouldn't let call `.preventDefault` on `undefined`  --> I didn't pass it an event, I'm not sure what would have constituted an event or if I could have just passed it null or something.

Reply:
The event would be your click events in LoginButton and LogoutButton in SessionButton.js. To be honest I'm confused about why you'd get that error, but if it's not letting you call it then I'd try leaving it out. 
The reason event.preventDefault() is used is to prevent the default action of an element from occuring on the event. In this case it should prevent the button that has the click event from refreshing the page.